src/Common/Telemetry/Telemetry.php

```diff
@@ -0,0 +1,458 @@
+	 */
+	public static function get_terms_url(): string {
+		return esc_url( apply_filters( 'tec_common_telemetry_terms_url', 'https://evnt.is/1bcm' ) );
+	}
+
+	/**
+	 * Get the URL for the Privacy Policy link in the optin modal.
+	 *
+	 * @since TBD
+	 *
+	 * @return string
+	 */
+	public static function get_privacy_url(): string {
+		return esc_url( apply_filters( 'tec_common_telemetry_privacy_url', 'https://evnt.is/1bcn' ) );
+	}
+
+	/**
+	 * Filters the exit questionnaire shown during plugin deactivation/uninstall.
+	 *
+	 * @since TBD
+	 *
+	 * @param array<string,mixed> $args The current args.
+	 *
+	 * @return array<string,mixed> $args The modified args.
+	 */
+	public function filter_exit_interview_args( $args ) {
+		$new_args = [
+			'plugin_logo'        => tribe_resource_url( 'images/logo/tec-brand.svg', false, null, \Tribe__Main::instance() ),
+			'plugin_logo_width'  => 'auto',
+			'plugin_logo_height' => 32,
+			'plugin_logo_alt'    => 'TEC Common Logo',
+			'heading'            => __( 'We’re sorry to see you go.', 'tribe-common' ),
+			'intro'              => __( 'We’d love to know why you’re leaving so we can improve our plugin.', 'tribe-common' ),
+			'uninstall_reasons'  => [
+				[
+					'uninstall_reason_id' => 'confusing',
+					'uninstall_reason'    => __( 'I couldn’t understand how to make it work.', 'tribe-common' ),
+				],
+				[
+					'uninstall_reason_id' => 'better-plugin',
+					'uninstall_reason'    => __( 'I found a better plugin.', 'tribe-common' ),
+					'show_comment'        => true,
+				],
+				[
+					'uninstall_reason_id' => 'no-feature',
+					'uninstall_reason'    => __( 'I need a specific feature it doesn’t provide.', 'tribe-common' ),
+					'show_comment'        => true,
+				],
+				[
+					'uninstall_reason_id' => 'broken',
+					'uninstall_reason'    => __( 'The plugin doesn’t work.', 'tribe-common' ),
+					'show_comment'        => true,
+				],
+				[
+					'uninstall_reason_id' => 'other',
+					'uninstall_reason'    => __( 'Other', 'tribe-common' ),
+					'show_comment'        => true,
+				],
+			],
+		];
+
+		return array_merge( $args, $new_args );
+	}
+
+	/**
+	 * Triggers Telemetry's opt-in modal with our parameters.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function show_optin_modal(): void {
+		$plugin_slug = self::get_plugin_slug();
+
+		/**
+		 * Filter allowing disabling of the optin modal.
+		 * Returning boolean false will disable the modal
+		 *
+		 * @since TBD
+		 *
+		 * @param bool $show Whether to show the modal or not.
+		 */
+		$show = (bool) apply_filters( 'tec_common_telemetry_show_optin_modal', true );
+
+		if ( ! $show ) {
+			return;
+		}
+
+		do_action( 'stellarwp/telemetry/optin', $plugin_slug );
+	}
+
+	/**
+	 * Saves the "Opt In Status" setting.
+	 *
+	 * @return void
+	 */
+	public function save_opt_in_setting_field(): void {
+		// Return early if not saving the Opt In Status field.
+		if ( ! isset( $_POST[ 'current-settings-tab' ] ) ) {
```

Comment:
Any particular reason we are not using `tribe_get_request_var()` here, saving on a variable and using it down on line 363?